src/utp_stream.cpp

```diff
@@ -1524,6 +1543,24 @@ std::size_t utp_socket_impl::available() const
 	return m_receive_buffer_size;
 }
 
+
+void utp_socket_impl::decrease_receive_buffer(int const bytes)
+{
```

Comment:
as far as I can tell, all this function should do is to increment ``m_receive_buffer_surplus`` by ``bytes``. The actual shrinkage of ``m_receive_buffer_capacity`` must be deferred.

Reply:
I see now. the ``bytes`` arguments essentially is how much we're allowed to decrement the capacity, i.e. the number of bytes consumed by the upper layer.